### PR TITLE
Adding Integration tests for empty body requests with Content aware mediation

### PR DESCRIPTION
**Pull request by Milanka00**
## Purpose

This PR adds a integration test for POST/PUT/PATCH/GET/DELETE requests to validate that API invocation do not fail when requests contain:

- Content-Type: application/json
- an empty request body
- a content-aware Synapse mediator using json-eval(...)

## Related issues

- https://github.com/wso2/api-manager/issues/4826

## Related fix

- https://github.com/wso2/wso2-synapse/pull/2542

### CODE REVIEW

File: all-in-one-apim/modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/operationPolicy/ContentAwareMediationPolicyEmptyBodyTestCase.java [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+            case "PATCH":
+                request = new HttpPatch(invocationUrl);
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported verb: " + verb);
+        }
+        request.setHeader("Authorization", "Bearer " + accessToken);
+        request.setHeader("Content-Type", "application/json");
+
+        HttpClient client = HttpClientBuilder.create().setHostnameVerifier(new AllowAllHostnameVerifier()).build();
```
Comment (RakhithaRR) [inline all-in-one-apim/modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/operationPolicy/ContentAwareMediationPolicyEmptyBodyTestCase.java]: Can we use a single http client rather than creating one everytime? Also, lets close the client after test is done.

File: all-in-one-apim/modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/operationPolicy/ContentAwareMediationPolicyEmptyBodyTestCase.java [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    private String accessToken;
+    private String contentAwarePolicyId;
+
+    @BeforeClass(alwaysRun = true)
+    public void initialize() throws Exception {
+
+        super.init();
+        String gatewaySessionCookie = createSession(gatewayContextMgt);
+
+        loadSynapseConfigurationFromClasspath("artifacts" + File.separator + "AM" + File.separator
```
Comment (RakhithaRR) [inline all-in-one-apim/modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/operationPolicy/ContentAwareMediationPolicyEmptyBodyTestCase.java:92]: Do we need this in both base test and in this test?

**Review by RakhithaRR — APPROVED** [review]

**Review by tharindu1st — APPROVED** [review]
